New Control - Remote device picker docs (#174)
MicrosoftDocs/WindowsCommunityToolkitDocs: docs/helpers/RemoteDeviceHelper.md

---
title: RemoteDeviceHelper
author: avknaidu
description: The RemoteDeviceHelper is a Helper to get a List Remote Devices that are accessible.
keywords: windows 10, uwp, windows community toolkit, uwp community toolkit, uwp toolkit, RemoteDeviceHelper, helper
dev_langs:
  - csharp
---

# RemoteDeviceHelper 

The [RemoteDeviceHelper](https://docs.microsoft.com/dotnet/api/microsoft.toolkit.uwp.helpers.remotedevicehelper) gives you a list of Remote Systems. All the systems must be signed in with the same Microsoft Account (MSA)

> [!IMPORTANT]

> > Make sure you enable "RemoteSystem" Capability in `package.appxmanifest`

Reviewer: Can we keep the capability doc link here to be consistent with our CameraPreview format for this message?
```suggestion
> Make sure you enable the [RemoteSystem capability](https://docs.microsoft.com/en-us/windows/uwp/packaging/app-capability-declarations#general-use-capabilities) in your app's `package.appxmanifest` to access remote system information.
```

};

RemoteDeviceHelper _remoteDeviceHelper = new RemoteDeviceHelper(filters);
DevicesList.DataContext = _remoteDeviceHelper;

```

```xaml

<ListView ItemsSource="{Binding RemoteSystems}" x:Name="DevicesList">
  <ListView.ItemTemplate>
    <DataTemplate>
	  <Grid>
		<Grid.RowDefinitions>
		  <RowDefinition Height="*"/>
		  <RowDefinition Height="*"/>
		</Grid.RowDefinitions>
		<TextBlock Text="{Binding DisplayName}" Tag="{Binding }" Grid.Row="0" />
		<TextBlock Text="{Binding ModelDisplayName}" Tag="{Binding }" Grid.Row="1" />
	  </Grid>
    </DataTemplate>
  </ListView.ItemTemplate>
</ListView>

```

## Sample Code

[RemoteDeviceHelper Sample Page Source](https://github.com/windows-toolkit/WindowsCommunityToolkit/tree/master/Microsoft.Toolkit.Uwp.SampleApp/SamplePages/RemoteDeviceHelper). You can see this in action in [Windows Community Toolkit Sample App](https://www.microsoft.com/store/apps/9NBLGGH4TLCQ).

## Requirements

| Device family | Universal, 10.0.16299.0 or higher |
| --- | --- |
| Namespace | Microsoft.Toolkit.Uwp.Helpers |
| NuGet package | [Microsoft.Toolkit.Uwp](https://www.nuget.org/packages/Microsoft.Toolkit.Uwp/) |

## API

> * [RemoteDeviceHelper source code](https://github.com/Microsoft/UWPCommunityToolkit/tree/master/Microsoft.Toolkit.Uwp/Helpers/RemoteDeviceHelper)

Reviewer: ```suggestion
* [RemoteDeviceHelper source code](https://github.com/windows-toolkit/WindowsCommunityToolkit/tree/master/Microsoft.Toolkit.Uwp/Helpers/RemoteDeviceHelper)
```

---
title: RemoteDeviceHelper
author: avknaidu
description: The RemoteDeviceHelper is a Helper to get a List Remote Devices that are accessible.
keywords: windows 10, uwp, windows community toolkit, uwp community toolkit, uwp toolkit, RemoteDeviceHelper, helper

> ---

Reviewer: ```suggestion
dev_langs:
  - csharp
---
```

# RemoteDevicePicker Control 

The [RemoteDeviceHelper](https://docs.microsoft.com/dotnet/api/microsoft.toolkit.uwp.helpers.remotedevicepicker) gives you a list of Remote Systems. All the systems must be signed in with the same Microsoft Account (MSA)

> **Make sure you enable "RemoteSystem" Capability in `package.appxmanifest`**

Reviewer: We should call this out with the important syntax:
```suggestion
> [!IMPORTANT]
> Make sure you enable the [RemoteSystem capability](https://docs.microsoft.com/en-us/windows/uwp/packaging/app-capability-declarations#general-use-capabilities) in your app's `package.appxmanifest` to access remote system information.
```

Reviewer: @avknaidu can you include the capabilities doc page link here still?  Thanks!

## Syntax

```c#

RemoteDeviceHelper _remoteDeviceHelper = new RemoteDeviceHelper();

```

> You can also use default filer types for initailizing. Like Below.

Reviewer: ```suggestion
You can also use default filter types for initializing. Like Below.
```

```c#

var filters = new List<IRemoteSystemFilter>
{
    new RemoteSystemDiscoveryTypeFilter(RemoteSystemDiscoveryType.Cloud),
    new RemoteSystemAuthorizationKindFilter(RemoteSystemAuthorizationKind.SameUser),
    new RemoteSystemStatusTypeFilter(RemoteSystemStatusType.Available)
};

RemoteDeviceHelper _remoteDeviceHelper = new RemoteDeviceHelper(filters);
```

> ## ## Example

Reviewer: Extra ## here:
```suggestion
## ## Example
```

```c#
// without filters

RemoteDeviceHelper _remoteDeviceHelper = new RemoteDeviceHelper();
DevicesList.DataContext = _remoteDeviceHelper;

// with filters

var filters = new List<IRemoteSystemFilter>
{
    new RemoteSystemDiscoveryTypeFilter(RemoteSystemDiscoveryType.Cloud),
    new RemoteSystemAuthorizationKindFilter(RemoteSystemAuthorizationKind.SameUser),
    new RemoteSystemStatusTypeFilter(RemoteSystemStatusType.Available)
};

RemoteDeviceHelper _remoteDeviceHelper = new RemoteDeviceHelper(filters);
DevicesList.DataContext = _remoteDeviceHelper;

```

```xaml

<ListView ItemsSource="{Binding RemoteSystems}" x:Name="DevicesList">
  <ListView.ItemTemplate>
    <DataTemplate>
	  <Grid>
		<Grid.RowDefinitions>
		  <RowDefinition Height="*"/>
		  <RowDefinition Height="*"/>
		</Grid.RowDefinitions>
		<TextBlock Text="{Binding DisplayName}" Tag="{Binding }" Grid.Row="0" />
		<TextBlock Text="{Binding ModelDisplayName}" Tag="{Binding }" Grid.Row="1" />
	  </Grid>
    </DataTemplate>
  </ListView.ItemTemplate>
</ListView>

```

## Sample Code

> [RemoteDeviceHelper Sample Page Source](https://github.com/Microsoft/UWPCommunityToolkit/tree/master/Microsoft.Toolkit.Uwp.SampleApp/SamplePages/RemoteDeviceHelper). You can see this in action in [Windows Community Toolkit Sample App](https://www.microsoft.com/store/apps/9NBLGGH4TLCQ).

Reviewer: ```suggestion
[RemoteDeviceHelper Sample Page Source](https://github.com/windows-toolkit/WindowsCommunityToolkit/tree/master/Microsoft.Toolkit.Uwp.SampleApp/SamplePages/RemoteDeviceHelper). You can see this in action in [Windows Community Toolkit Sample App](https://www.microsoft.com/store/apps/9NBLGGH4TLCQ).
```

## Requirements

| Device family | Universal, 10.0.16299.0 or higher |
| --- | --- |
| Namespace | Microsoft.Toolkit.Uwp.Helpers |
| NuGet package | [Microsoft.Toolkit.Uwp.Helpers](https://www.nuget.org/packages/Microsoft.Toolkit.Uwp.Helpers/) |

> 

Reviewer: Link to API source?
